src/Features/Features.php

```diff
@@ -0,0 +1,268 @@
+
+use Automattic\WooCommerce\Admin\Loader;
+
+/**
+ * Features Class.
+ */
+class Features {
+	/**
+	 * Class instance.
+	 *
+	 * @var Loader instance
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Get class instance.
+	 */
+	public static function get_instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		// Load feature before WooCommerce update hooks.
+		add_action( 'init', array( __CLASS__, 'load_features' ), 4 );
+		add_filter( 'woocommerce_get_sections_advanced', array( __CLASS__, 'add_features_section' ) );
+		add_filter( 'woocommerce_get_settings_advanced', array( __CLASS__, 'add_features_settings' ), 10, 2 );
+		add_filter( 'woocommerce_get_settings_advanced', array( __CLASS__, 'maybe_load_beta_features_modal' ), 10, 2 );
+		add_action( 'admin_enqueue_scripts', array( __CLASS__, 'load_scripts' ), 15 );
+		add_filter( 'admin_body_class', array( __CLASS__, 'add_admin_body_classes' ) );
+		add_filter( 'update_option_woocommerce_allow_tracking', array( __CLASS__, 'maybe_disable_features' ), 10, 2 );
+	}
+
+	/**
+	 * Gets a build configured array of enabled WooCommerce Admin features/sections.
+	 *
+	 * @return array Enabled Woocommerce Admin features/sections.
+	 */
+	public static function get_features() {
+		return apply_filters( 'woocommerce_admin_features', array() );
+	}
+
+	/**
+	 * Returns if a specific wc-admin feature is enabled.
+	 *
+	 * @param  string $feature Feature slug.
+	 * @return bool Returns true if the feature is enabled.
+	 */
+	public static function is_enabled( $feature ) {
+		$features = self::get_features();
+		return in_array( $feature, $features, true );
+	}
+
+	/**
+	 * Get the feature class as a string.
+	 *
+	 * @param string $feature Feature name.
+	 * @return string|null
+	 */
+	public static function get_feature_class( $feature ) {
+		$feature       = str_replace( '-', '', ucwords( strtolower( $feature ), '-' ) );
+		$feature_class = 'Automattic\\WooCommerce\\Admin\\Features\\' . $feature;
+
+		if ( class_exists( $feature_class ) ) {
+			return $feature_class;
+		}
+
+		// Handle features contained in subdirectory.
+		if ( class_exists( $feature_class . '\\Init' ) ) {
+			return $feature_class . '\\Init';
+		}
+
+		return null;
+	}
+
+	/**
+	 * Class loader for enabled WooCommerce Admin features/sections.
+	 */
+	public static function load_features() {
+		$features = self::get_features();
+		foreach ( $features as $feature ) {
+			$feature_class = self::get_feature_class( $feature );
+
+			if ( $feature_class ) {
+				new $feature_class();
+			}
+		}
+	}
+
+	/**
+	 * Disable a toggleable feature.
+	 *
+	 * @param string $feature Feature name.
+	 * @return bool
+	 */
+	public static function disable( $feature ) {
+		$feature_class = self::get_feature_class( $feature );
+
+		if ( $feature_class && defined( "$feature_class::TOGGLE_OPTION_NAME" ) ) {
```

Comment:
So, as it is, any feature class with this `TOGGLE_OPTION_NAME` defined will be disabled with tracking?

Reply:
That's right.  We'd want to opt out of any toggleable features (presumably all toggleable features are beta features) so this helps with other features going forward.

We may want to rename to `beta_features` as opposed to `toggleable_features` so it's clear that these will be turned off when tracking is opted out.